### ISSUE TITLE
Is it possible to run node related queries instead of relation related queries?

### ISSUE THREAD

**Issue by MrSage** [aplbrain/grand-cypher#8]
I apologize for asking this from a place of relatively little understanding. I am working with a project which has brought me further into graphs than I have ever ventured before. For my project I need a way to query my graph for nodes which match a pattern. For the moment I am using https://geronimo-iia.github.io/networkx-query/ . However, I am curious to use this library for the expansive capabilities of `cypher`. Is it possible to make this query work within the supported syntax of the library?

```
MATCH (c:City)
WHERE c.name = "London"
RETURN c
```

I'm reading through the source code to try to answer this for myself but there are a lot of new concepts being introduced to me all at once so I figured it wouldn't hurt to ask. I appreciate the time and effort!

**Comment by j6k4m8** [comment]
Hi @MrSage! 

The short answer is YES: You can use this tool to perform this type of search, for sure! 

Right now, we don't support different node tags, so the `:City` component of that query won't parse. In the meantime, you can do something like this:

```js
MATCH (c)
WHERE c.name = "London"
RETURN c
```

The one stipulation here is that we use [GrandIso](https://GitHub.com/aplbrain/grandiso-networkx) under the hood here, so you can't search for multiple [connected components](https://en.wikipedia.org/wiki/Component_(graph_theory)). That is, you CANNOT run this search currently:

```js
MATCH (b)
MATCH (c)
WHERE c.name = "London" AND b.name = "NYC"
RETURN b, c
```

Because `b` and `c` represent two different connected components in the query. 

You CAN, however, run this search:

```js
MATCH (c)-[]->(b)
WHERE c.name = "London"
AND b.type = "City"
RETURN b, c
```

because the search is ONE connected component.

The other thing to consider here is that you might not need a graph search tool if you're not looking for edges: Looking for specific nodes might be way easier in a "plain old" search like this (assuming a networkx.Graph):

```python
node_matches = []

for node_id, attributes in my_graph.nodes(data=True):
    if attributes['name'] == "London":
        node_matches.append(node_id)
```

But maybe I'm misunderstanding your use-case? Looking forward to hearing if this helps!

**Comment by MrSage** [comment]
@j6k4m8 , thank you so much for taking the time to answer! I guess I must have had some weird environment issues. I was able to get the queries you provided to work when I reinstalled the library. 

I was seeing this error every time I tried to run a query like that:

>E           lark.exceptions.UnexpectedCharacters: No terminal matches 'W' in the current parser context, at line 3 col 1
E           
E           WHERE A.name = "London"
E           ^
E           Expected one of: 
E               * MINUS

I must have been on an old or corrupted version of the library somehow. I _think_ I'll need to query on edges in the future so I wanted to pick something now that would be able to support both styles. And having a google-able query language helps a lot for my own sanity 😇 

Thank you guys so much for this library!
